### ISSUE TITLE
Don't highlight during insert. 

### ISSUE THREAD

**Issue by jorundoa** [joshestein/ideavim-quickscope#7]
Is it possible to disable the highlighting effect during insert mode? As this mode doesn't allow the usage of f/t anway, and it's kind of distracting that the letters keep changing colours as i write.

**Comment by joshestein** [comment]
Hey! Thank you for the suggested feature.

I assume this is for when no keys have been specified for triggers for quickscope? That is, `g:qs_highlight_on_keys` is not set?

I had a quick look into how to achieve this. Unfortunately, we do not listen to mode changes, but rather caret event changes. Switching from normal to insert mode does not trigger a caret change event. This means that you if you enter insert mode, the highlights would persist until you press some other key. Similarly, when you exit insert mode no new highlights will be shown until you move the cursor again.

https://user-images.githubusercontent.com/27491286/222729644-9b4c6d34-8eac-4605-a412-047a6ef60a42.mov

What do you think?

**Comment by jorundoa** [comment]
Actually, I couldn't get `g:qs_highlight_on_keys` to work either. 
It shouldn't be more than a 
`let g:qs_highlight_on_keys = ['f', 'F', 't', 'T'] ` in ideavimrc right? For some reason the highlights always show...

**Comment by joshestein** [comment]
Yes, that's right. Did you try restarting your IDE after adding the line to your ideavimrc?

**Comment by jorundoa** [comment]
Got it working! Had to reset a cache or two apparently. 

As for the issue, I believe this should solve it for me. However if someone doesn't want to rely on the highlight on keys functionality (I haven't quite decided what I prefer to be honest) as its the default functionality in normal quick-scope, this would probably bug them quite a bit

**Comment by joshestein** [comment]
Great to hear you got it working 😊 

Yes, you are right - the constant highlights are probably annoying. Will publish a new version to the Jetbrains marketplace soonest.

**Comment by joshestein** [comment]
Closing as I've pushed the above fix to the marketplace.

Unfortunately until [autocommands are implemented](https://youtrack.jetbrains.com/issue/VIM-1693/Add-support-for-autocmd) and we can listen to the `InsertEnter` event we are manually inspecting the vim mode in the caret listener. The current solution is not ideal - we have a delay between going into insert mode and typing some letter before removing existing highlights.
